File: python/1710.py
class Solution:
    def maximumUnits(self, boxTypes: List[List[int]], truckSize: int) -> int:
        boxTypes.sort(key = lambda x: x[1], reverse = True)
        cur_unit = 0
        n = len(boxTypes)
        for i in range(n):
            b, u = boxTypes[i]
            if b <= truckSize:
                truckSize -= b
                cur_unit += b * u
            else:
                cur_unit += truckSize * u
                truckSize = 0 
        return cur_unit

File: java/1710.java
class Solution {
    public int maximumUnits(int[][] boxTypes, int truckSize) {
        Arrays.sort(boxTypes, (a,b) ->  (b[1] - a[1]) );
        int n = boxTypes.length;
        int res = 0;
        for (int i = 0; i < n; i ++){
            int box  = boxTypes[i][0];
            int unit = boxTypes[i][1];
            if (box <= truckSize){
                res += box * unit;
                truckSize -= box;
            }else{
                res += truckSize * unit;
                break;
            }
        }
        return res;
        
    }
}
